doc/operations.md

```diff
@@ -0,0 +1,64 @@
+# Thoth Operations
+
+The following sections describe how we would like to handle certain categories of components of applications deployed on OpenShift.
+
+We focus on provision/deprovision use cases first. Later additions to this document may take other operational aspects into account.
+
+## Builder ImageStreams
+
+Builder ImageStreams are considered common requirements. They should not be created or deleted by an application (de)provisioning playbook. A provision playbook should use the ansible role [thoth-core-imagestreams](https://galaxy.ansible.com/thoth-station/thoth-core-imagestreams) to set up Builder ImageStreams.
+
+## Application Components
+
+Application Components, such as Deployments, ImageStreams, CronJobs, should be created and deleted by (de)provision playbooks. Both should be idempotent.
+
+### Persistent Volumes and their Claims
+
+If persistent volumes are required by an application the corresponding Persistant Volume Claims should be created by the provision playbook, but not be removed by the deprovision playbook. We want to preserve any data that has been created!
+
+If a Persistent Volume Claim already exists, it must not be changed by the provision playbook.
+
+## OpenShift Templates
+
+OpenShift Template should be created for
+
+* Application ImageStreams
+* Application components needed for deployment
+* Application components needed for building
+
+All OpenShift objects must carry application and component labels. 
+
+### Example:
+
+```yaml
+  - apiVersion: v1
+    kind: BuildConfig
+    metadata:
+      name: adviser
+      labels:
+        app: thoth
+        component: adviser
+    spec:
+     ...
+```
+
+The meta data for the OpenShift Template itself must carry more annotations:
+
+```yaml
+apiVersion: v1
+kind: Template
+metadata:
+  name: adviser-buildconfig
+  annotations:
+    description: This is Thoth Adviser BuildConfig...
+    openshift.io/display-name: Thoth Adviser BuildConfig
+    version: 0.3.0
```

Comment:
I would completely remove versions from templates. IMHO we should keep only one version - that is the version of application.

Reply:
well, its the version of the template, it might be a quick way to figure out what version of the template is instantiated on OpenShift. There is no other way to trace back to a git commit hash

Reply:
The thing is that these versions do not correspond to application version and they are harder to track/update so one needs to anyway check based on git log what has changed. I would rather do releases on github where we pack template and say "this is version X.Y and this template is up2date template how you deploy this application".

Reply:
agree on the github release containing a template, but that doesnt help with the problem that I want to figure out which version of the template has been created on OpenShift...

Reply:
Do we need that information? If I run a component that does not deploy due to template issues, I know which version of template I should use instead.

Reply:
hmm, fair ...